### PR TITLE
zh-cn: update `rel=preload` page

### PR DESCRIPTION
**Pull request by codersjj**
### Description

Update `rel=preload` page.

### Motivation

Synchronize English document content.

### Additional details

See [#27738](https://github.com/mdn/content/pull/27738)

### CODE REVIEW

**Comment by yin1999** [comment]
Could you fix the errors listed [here](https://github.com/mdn/translated-content/actions/runs/6281961500/job/17060886469?pr=16030)?

**Comment by codersjj** [comment]
> Could you fix the errors listed [here](https://github.com/mdn/translated-content/actions/runs/6281961500/job/17060886469?pr=16030)?

image
I can't see the error details. What should I do?

**Comment by codersjj** [comment]
> image
> 
> Line 11-25, spacing and punctuating

Thanks! It seems my internet had some issues before, but it's fine now.

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 1]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+  <canvas></canvas>
+
+  <script src="main.js" defer></script>
+</body>
+```
+
+在这里，我们预加载我们的 CSS 和 JavaScript 文件，以便它们在稍后渲染页面时立即可用。这个示例可能过于简单，因为浏览器很可能在与预加载的 HTML 块中相同的位置发现了 `<link rel="stylesheet">` 和 `<script>` 元素，但在稍后发现资源并且资源越大时，好处就会更加明显。例如：
+
+- 从 CSS 内部指向的资源，如字体或图像。
+- JavaScript 可以请求的资源，如 JSON、导入的脚本或 Web Workers。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:47]: removing 's'

```suggestion
- JavaScript 可以请求的资源，如 JSON、导入的脚本或 Web Worker。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 2]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+- 较大的图像和视频文件。
+
+`preload` 还有其他优点。使用 `as` 来指定要预加载的内容类型允许浏览器：
+
+- 更准确地优先加载资源。
+- 将资源存储在缓存中以备将来使用，在适当的情况下复用资源。
+- 对资源应用正确的[内容安全策略](/zh-CN/docs/Web/HTTP/CSP)。
+- 为其设置正确的 {{HTTPHeader("Accept")}} 请求头。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:55]:
```suggestion
- 为其设置正确的 {{HTTPHeader("Accept")}} 请求标头。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 3]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+- `embed`: 用于嵌入在 {{htmlelement("embed")}} 元素中的资源。
+- `fetch`: 通过 fetch 或 XHR 请求访问的资源，例如 ArrayBuffer、WebAssembly 二进制文件或 JSON 文件。
+- `font`: 字体文件。
+- `image`: 图像文件。
+- `object`: 要嵌入在 {{htmlelement("object")}} 元素中的资源。
+- `script`: JavaScript 文件。
+- `style`: CSS 样式表。
+- `track`: WebVTT 文件。
+- `worker`: JavaScript web worker 或 shared worker。
+- `video`: 视频文件，通常在 {{htmlelement("video")}} 中使用。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:72]:
```suggestion
- `audio`：音频文件，通常在 {{htmlelement("audio")}} 中使用。
- `document`：用于嵌入在 {{htmlelement("frame")}} 或 {{htmlelement("iframe")}} 中的 HTML 文档。
- `embed`：用于嵌入在 {{htmlelement("embed")}} 元素中的资源。
- `fetch`：通过 fetch 或 XHR 请求访问的资源，例如 ArrayBuffer、WebAssembly 二进制文件或 JSON 文件。
- `font`：字体文件。
- `image`：图像文件。
- `object`：要嵌入在 {{htmlelement("object")}} 元素中的资源。
- `script`：JavaScript 文件。
- `style`：CSS 样式表。
- `track`：WebVTT 文件。
- `worker`：JavaScript web worker 或 shared worker。
- `video`：视频文件，通常在 {{htmlelement("video")}} 中使用。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 4]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+
+> **注意：** 预加载 `font` 和 `fetch` 资源需要设置 `crossorigin` 属性；请参阅下文的[启用 CORS 的获取请求](#启用_CORS_的获取请求)。
+
+> **注意：** 关于这些值以及它们所期望被使用的 web 功能的更多细节可以在预加载规范中找到 — 请参见[链接元素扩展](https://w3c.github.io/preload/#link-element-extensions)。此外，请注意，`as` 属性可以接受的值的完整列表受 Fetch 规范的约束 — 请参见[请求目标](https://fetch.spec.whatwg.org/#concept-request-destination)。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:76]: Note card highlight: https://developer.mozilla.org/zh-CN/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#%E7%A4%BA%E4%BE%8B

```suggestion
> **备注：** 预加载 `font` 和 `fetch` 资源需要设置 `crossorigin` 属性；请参阅下文的[启用 CORS 的获取请求](#启用_cors_的获取请求)。

> **备注：** 关于这些值以及它们所期望被使用的 web 功能的更多细节可以在预加载规范中找到——请参见[链接元素扩展](https://w3c.github.io/preload/#link-element-extensions)。此外，请注意，`as` 属性可以接受的值的完整列表受 Fetch 规范的约束——请参见[请求目标](https://fetch.spec.whatwg.org/#concept-request-destination)。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 5]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+
+## 包括 MIME 类型
+
+`<link>` 元素可以接受一个 [`type`](/zh-CN/docs/Web/HTML/Element/link#type) 属性，其中包含元素指向的资源的 MIME 类型。这在预加载资源时特别有用 — 浏览器将使用 `type` 属性的值来确定是否支持该资源，如果不支持，则会忽略它，仅在支持时才会下载它。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:80]:
```suggestion
`<link>` 元素可以接受一个 [`type`](/zh-CN/docs/Web/HTML/Element/link#type) 属性，其中包含元素指向的资源的 MIME 类型。这在预加载资源时特别有用——浏览器将使用 `type` 属性的值来确定是否支持该资源，如果不支持，则会忽略它，仅在支持时才会下载它。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 6]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+      Your browser doesn't support HTML video. Here is a
+      <a href="sintel-short.mp4">link to the video</a> instead.
+    </p>
+  </video>
+</body>
+```
+
+上面的示例代码只会在支持的浏览器中预加载 `video/mp4` 格式的视频，对于那些浏览器中支持 `video/mp4` 格式的用户，实际上会使用 `video/mp4` 格式的视频（因为它是首个指定的 {{htmlelement("source")}}）。这可以让拥有支持 `video/mp4` 格式的浏览器的用户获得更流畅和更响应的视频播放体验。
+
+请注意，对于那些浏览器既支持 `video/mp4` 格式又支持 `video/webm` 格式的用户，如果在上述代码中还指定了一个 `<link rel="preload" href="sintel-short.webm" as="video" type="video/webm">` 元素，那么两种格式的视频（`video/mp4` 和 `video/webm`）_都_ 会被预加载，尽管实际上只有其中一种会被使用。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:105]: removing the space

```suggestion
请注意，对于那些浏览器既支持 `video/mp4` 格式又支持 `video/webm` 格式的用户，如果在上述代码中还指定了一个 `<link rel="preload" href="sintel-short.webm" as="video" type="video/webm">` 元素，那么两种格式的视频（`video/mp4` 和 `video/webm`）*都*会被预加载，尽管实际上只有其中一种会被使用。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 7]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+
+然而，缺少预加载并不会阻止那些需要的用户实际使用 `video/webm` 视频：对于那些浏览器不支持 `video/mp4` 但支持 `video/webm` 的用户，上述示例中的代码仍然会使 `video/webm` 视频被使用——但它这样做的同时也不会导致对大多数其他用户不必要地进行预加载。
+
+## 启用 CORS 的获取请求
+
+在预加载启用 [CORS](/zh-CN/docs/Web/HTTP/CORS) 的资源（例如 [`fetch()`](/zh-CN/docs/Web/API/fetch)、[`XMLHttpRequest`](/zh-CN/docs/Web/API/XMLHttpRequest) 或[字体](/zh-CN/docs/Web/CSS/@font-face)）时，需要特别注意在你的 [`<link>`](/zh-CN/docs/Web/HTML/Element/link) 元素上设置 [`crossorigin`](/zh-CN/docs/Web/HTML/Element/link#crossorigin) 属性。该属性需要设置为与资源的 CORS 和凭据模式相匹配，即使获取请求不跨域也需要设置。
+
+如上所述，其中一个适用的有趣情况是字体文件。由于各种原因，这些文件必须使用匿名模式的 CORS 进行获取（参见[字体获取要求](https://drafts.csswg.org/css-fonts/#font-fetching-requirements)）。
+
+Let's use this case as an example. You can see the full [example source code on GitHub](https://github.com/mdn/html-examples/tree/master/link-rel-preload/fonts) ([also see it live](https://mdn.github.io/html-examples/link-rel-preload/fonts/)):
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:117]:
```suggestion
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 8]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+    }
+  </script>
+</body>
+```
+
+我们在 `<link>` 元素上包含 `media` 属性，以便在用户的视口较窄时预加载窄图像，在视口较宽时加载宽图像。我们使用 {{domxref("Window.matchMedia")}} / {{domxref("MediaQueryList")}} 来实现这一点（查看[测试媒体查询](/zh-CN/docs/Web/CSS/CSS_media_queries/Testing_media_queries)以获取更多信息）。
+
+这使得字体在页面渲染时更有可能可用，减少了 FOUT（未经样式处理的文本闪烁）。
+
+这并不局限于图像，甚至不局限于同类型的文件 —— 大胆想！如果用户使用的是带宽和 CPU 可能更受限的窄屏幕，你可以预加载和显示一个简单的 SVG 图表，或者如果用户的资源更丰富，可以预加载一个复杂的 JavaScript 代码块，然后用它来渲染一个交互式的 3D 模型。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:193]:
```suggestion
这并不局限于图像，甚至不局限于同类型的文件——大胆想象！如果用户使用的是带宽和 CPU 可能更受限的窄屏幕，你可以预加载和显示一个简单的 SVG 图表，或者如果用户的资源更丰富，可以预加载一个复杂的 JavaScript 代码块，然后用它来渲染一个交互式的 3D 模型。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 9]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+
+## 脚本和预加载
+
+> **注意：** 如果你正在使用 [JavaScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)，请使用 [`<link rel="modulepreload">`](/zh-CN/docs/Web/HTML/Attributes/rel/modulepreload)。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:197]:
```suggestion
> **备注：** 如果你正在使用 [JavaScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)，请使用 [`<link rel="modulepreload">`](/zh-CN/docs/Web/HTML/Attributes/rel/modulepreload)。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 10]
```diff
@@ -5,9 +5,225 @@ slug: Web/HTML/Attributes/rel/preload
...
+document.body.appendChild(preloadedScript);
+```
+
+当你想要预加载一个脚本，但需要将执行推迟到确切需要它的时候，这很有用。
+
+## 其他资源预加载机制
+
+还存在其他预加载特性，但都不如 `<link rel="preload">` 适合该目的：
+
+- `<link rel="prefetch">` 在浏览器中支持已久，但它是用于预获取将在下一次导航/页面加载时使用的资源（例如，当你跳转到下一页时）。这是可以的，但对于当前页面没有用！此外，浏览器会给预获取（`prefetch`）的资源比预加载（`preload`）的资源更低的优先级——当前页面比下一页更重要。有关更多详细信息，请参阅 {{Glossary("prefetch", "预获取")}}。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:223]:
```suggestion
- `<link rel="prefetch">` 在浏览器中支持已久，但它是用于预取将在下一次导航/页面加载时使用的资源（例如，当你跳转到下一页时）。这是可以的，但对于当前页面没有用！此外，浏览器会给预取（`prefetch`）的资源比预加载（`preload`）的资源更低的优先级——当前页面比下一页更重要。有关更多详细信息，请参阅{{Glossary("prefetch", "预取")}}。
```

File: files/zh-cn/web/html/attributes/rel/preload/index.md [hunk 11]
```diff
@@ -16,3 +232,7 @@ slug: Web/HTML/Attributes/rel/preload
 ## 浏览器兼容性
 
 {{Compat}}
+
+## 参见
+
+- [Preload: What Is It Good For?](https://www.smashingmagazine.com/2016/02/preload-what-is-it-good-for/) by Yoav Weiss
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/rel/preload/index.md:238]: Please translate the title

```suggestion
- [预加载的好处是什么？](https://www.smashingmagazine.com/2016/02/preload-what-is-it-good-for/)作者 Yoav Weiss
```

**Review by yin1999 — COMMENTED** [review]
some suggestions

**Comment by codersjj** [comment]
> some suggestions

Thanks a lot!

Looks good to me.

**Review by yin1999 — APPROVED** [review]
Thanks @codersjj!

**Pull request merged**